picard/ui/itemviews/custom_columns/manager_dialog.py

```diff
@@ -0,0 +1,302 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+"""Dialog to manage user-defined custom columns.
+
+This dialog lists all persisted custom columns and allows the user to add,
+edit, duplicate and delete entries. It delegates expression editing to
+``CustomColumnExpressionDialog`` for SRP / SOC.
+"""
+
+from __future__ import annotations
+
+from dataclasses import replace
+
+
+try:  # Allow import in non-GUI test environments
+    from PyQt6 import (
+        QtCore,
+        QtWidgets,
+    )
+except Exception:  # pragma: no cover - fallback for headless lint/test
+    QtCore = None  # type: ignore[assignment]
+    QtWidgets = None  # type: ignore[assignment]
+
+from picard.config import get_config
+from picard.i18n import gettext as _
+
+from picard.ui.itemviews.custom_columns.expression_dialog import CustomColumnExpressionDialog
+from picard.ui.itemviews.custom_columns.storage import (
+    CustomColumnSpec,
+    add_or_update_spec,
+    delete_spec_by_key,
+    load_specs_from_config,
+    save_specs_to_config,
+    unregister_and_delete,
+)
+
+
+class _SpecsTableModel(QtCore.QAbstractTableModel):
+    """Table model for custom column specs.
+
+    Parameters
+    ----------
+    specs
+        Initial list of specs to display.
+    parent
+        Parent QObject.
+    """
+
+    HEADERS = [
+        _("Field Name"),
+        _("Type"),
+        _("Expression"),
+        _("Align"),
+        _("Width"),
+    ]
+
+    def __init__(self, specs: list[CustomColumnSpec], parent: QtCore.QObject | None = None) -> None:
+        super().__init__(parent)
+        self._specs: list[CustomColumnSpec] = specs
+
+    # --- Model API ---------------------------------------------------------
+    def rowCount(self, parent: QtCore.QModelIndex | QtCore.QPersistentModelIndex | None = None) -> int:
+        return 0 if (parent and parent.isValid()) else len(self._specs)
+
+    def columnCount(self, parent: QtCore.QModelIndex | QtCore.QPersistentModelIndex | None = None) -> int:
+        return len(self.HEADERS)
+
+    def headerData(
+        self, section: int, orientation: QtCore.Qt.Orientation, role: int = QtCore.Qt.ItemDataRole.DisplayRole
+    ) -> object:
+        if orientation == QtCore.Qt.Orientation.Horizontal and role == QtCore.Qt.ItemDataRole.DisplayRole:
+            return self.HEADERS[section]
+        return None
+
+    def data(self, index: QtCore.QModelIndex, role: int = QtCore.Qt.ItemDataRole.DisplayRole) -> object:
+        if not index.isValid() or index.row() >= len(self._specs):
+            return None
+        spec = self._specs[index.row()]
+        if role == QtCore.Qt.ItemDataRole.DisplayRole:
+            col = index.column()
+            if col == 0:
```

Comment:
It would be great to not have hard-coded values here (the irony: Column was introduced partly to avoid that at the start).

It would be preferable to have an IntEnum for indexes and use a dict for `HEADERS`.